internal/services/containers/kubernetes_cluster_addons_resource_test.go

```diff
@@ -1144,3 +1159,267 @@ resource "azurerm_kubernetes_cluster" "test" {
+    }
+    x509_certificate_properties {
+      extended_key_usage = ["1.3.6.1.5.5.7.3.1"]
+      key_usage = [
+        "cRLSign",
+        "dataEncipherment",
+        "digitalSignature",
+        "keyAgreement",
+        "keyCertSign",
+        "keyEncipherment",
+      ]
+      subject_alternative_names {
+        dns_names = ["api.pluginsdk.io"]
+      }
+      subject            = "CN=api.pluginsdk.io"
+      validity_in_months = 1
+    }
+  }
+
+  depends_on = [azurerm_key_vault_access_policy.test]
+}
+
+resource "azurerm_key_vault_key" "test" {
+  name         = "testkeyvaultkey%[1]s"
+  key_vault_id = azurerm_key_vault.test.id
+
+  key_type = "RSA"
+  key_size = 2048
+
+  key_opts = [
+    "decrypt",
+    "encrypt",
+    "sign",
+    "unwrapKey",
+    "verify",
+    "wrapKey",
+  ]
+
+  depends_on = [azurerm_key_vault_access_policy.test]
+}
+
+resource "azurerm_user_assigned_identity" "test" {
+  name                = "acctestUAI-%[1]s"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%[1]s"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%[1]s"
+
+  linux_profile {
+    admin_username = "acctestuser%[1]s"
+
+    ssh_key {
+      key_data = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCqaZoyiz1qbdOQ8xEf6uEu1cCwYowo5FHtsBhqLoDnnp7KUTEBN+L2NxRIfQ781rxV6Iq5jSav6b2Q8z5KiseOlvKA/RF2wqU0UPYqQviQhLmW6THTpmrv/YkUCuzxDpsH7DUDhZcwySLKVVe0Qm3+5N2Ta6UYH3lsDf9R9wTP2K/+vAnflKebuypNlmocIvakFWoZda18FOmsOoIVXQ8HWFNCuw9ZCunMSN62QGamCe3dL5cXlkgHYv7ekJE15IA9aOJcM7e90oeTqo+7HTcWfdu0qQqPWY5ujyMw/llas8tsXY85LFqRnr3gJ02bAscjc477+X+j/gkpFoN1QEmt terraform@demo.tld"
+    }
+  }
+
+  default_node_pool {
+    name           = "default"
+    node_count     = 2
+    vm_size        = "Standard_DS2_v2"
+    vnet_subnet_id = azurerm_subnet.test.id
+    upgrade_settings {
+      max_surge = "10%%"
+    }
+  }
+
+  identity {
+    type = "UserAssigned"
+    identity_ids = [
+      azurerm_user_assigned_identity.test.id,
+    ]
+  }
+
+  network_profile {
+    network_plugin = "kubenet"
+    dns_service_ip = "10.10.0.10"
+    service_cidr   = "10.10.0.0/16"
+  }
+
+  key_vault_secrets_provider {
+    secret_rotation_enabled = true
+  }
+
+  service_mesh_profile {
+    mode                             = "Istio"
+    internal_ingress_gateway_enabled = true
+    external_ingress_gateway_enabled = true
+    certificate_authority {
+		key_vault_id           = azurerm_key_vault.test.id
+        root_cert_object_name  = azurerm_key_vault_certificate.test_cert1.name
+        cert_chain_object_name = azurerm_key_vault_certificate.test_cert2.name
+        cert_object_name       = azurerm_key_vault_certificate.test_cert3.name
+        key_object_name        = azurerm_key_vault_key.test.name
```

Comment:
Formatting looks a bit off here

Reply:
Thanks, all updated in the latest commit.